src/abstractdataframe/abstractdataframe.jl

```diff
@@ -842,11 +842,11 @@ without(df::AbstractDataFrame, c::Any) = without(df, index(df)[c])
 # catch-all to cover cases where indexing returns a DataFrame and copy doesn't
 
 Base.hcat(df::AbstractDataFrame, x; makeunique::Bool=false) =
-    hcat!(df[:, :], x, makeunique=makeunique)
+    hcat!(df[:], x, makeunique=makeunique)
```

Comment:
Using `copy` would be clearer? That's the intended effect here, right?

Maybe also add tests to ensure we don't change this behavior inadvertently?

Reply:
OK added tests and changed to `SubDataFrame`. Had to add a definition of `copy ` to `SubDataFrame`, but I guess it should have been implemented anyway.